test/blackbox-tests/test-cases/actions/concurrent.t

```diff
@@ -0,0 +1,107 @@
+Specification of the concurrency action:
+
+  $ cat > dune-project << EOF
+  > (lang dune 3.6)
+  > EOF
+
+  $ cat > dune << EOF
+  > (rule
+  >  (action
+  >   (concurrent )))
+  > EOF
+
+  $ dune build
+  File "dune", line 3, characters 2-15:
+  3 |   (concurrent )))
+        ^^^^^^^^^^^^^
+  Error: 'concurrent' is only available since version 3.7 of the dune language.
+  Please update your dune-project file to have (lang dune 3.7).
+  [1]
+
+Requires Dune 3.7.
+
+  $ cat > dune-project << EOF
+  > (lang dune 3.7)
+  > EOF
+
+It works just like (progn ):
+
+  $ dune build
+  File "dune", line 1, characters 0-32:
+  1 | (rule
+  2 |  (action
+  3 |   (concurrent )))
+  Error: Rule has no targets specified
+  [1]
+
+  $ cat > dune << EOF
+  > (rule
+  >  (action
+  >   (concurrent
+  >    (progn (system "") (echo "A\n"))
+  >    (progn (echo "B\n"))
+  >    (with-outputs-to some-target (progn) ))))
+  > EOF
+
+But runs the actions concurrently. Seen here by adding some delay to the
+printing of A.
```

Comment:
Seems pretty flaky: such timing assumptions can easily get violated, e.g. when the CI machine is busy.

Reply:
If you'd like to verify concurrency, you better use some communication mechanism.

Reply:
Yeah, it's super flaky. I don't have any experience testing using a communication mechanism. Could you elaborate?

Reply:
Not sure what would be best/most idiomatic for Dune tests. @rgrinberg Any suggestions?

Reply:
Honestly, I wouldn't bother with this test at all. I trust our implementation of `Fiber.parallel_map`.

But if you insist, I suppose you could do the following:

```
(progn
 (run mkfifo a b)
 (concurrent (run ./run.sh a b) (run ./run.sh b a)))
```

With run.sh being:

```
echo foo >$1
read line <$2
```

The above action cannot terminate with `-j1`.

Reply:
I think I'd prefer to have a test like this. It seems important to test that `concurrent` is, well, concurrent.

I like your suggestion!

Reply:
I think I'd prefer to have a test like this. It seems important to test that `concurrent` is, well, concurrent.

I like your suggestion!